### ISSUE TITLE
[Feature Request]: enhance UX on image import, replace "Registry"

### ISSUE THREAD

**Issue by ikke-t** [opendatahub-io/odh-dashboard#864]
### Feature description

There is one confusing dialog that has caused misery at customer and partner.
It's small change, but UI is a bit misleading ATM. I've seen failures to import image in different ways, as the term "Repository" and "Name" are misleading. This dialog:

![Screenshot from 2022-12-16 11-07-17](https://user-images.githubusercontent.com/1146409/208070208-4d3f6774-1376-4ca2-9581-51b80b8bd768.png)

First of all, the word repository typically means server url like quay.io, registry.access.redhat.com. So people put those there, and enter the name of image like ```image:tag``` into the "name" field, as it can be read that way. Another error I've seen is to put http:// or https:// to "Repository" field.

### Describe alternatives you've considered

In minimum, I'd change the Repository -text to "Image URL". Then I'd have an example there, like:

```Image URL, e.g.: repository.com:5000/path/image:tag```

### Anything else?

I've heard from Max Murakami in slack that actually the example of repository.com:5000/path/image:tag is bad, as there is a bug that the dialog will grab anything after first colon as tag. That should be fixed while at it, the regexp should only take anything after the last colon as tag.

**Comment by shalberd** [comment]
Agreed that the grayed-out ddefault value of that field or a hover-over popup needs to be more clear on what that field is doing, formats, all that.

I wonder also how you are dealing with importing from the openshift-internal registry, as that requires custom CAs often ... unless you use no verification of trust chain. Same for importing from e.g. a VMWare Harbor or Artifactory Docker Repo. Regarding the naming: I would neither call it repository nor Image Url, but docker image path. On a whole, I have not used this before, just making you aware that importing from a public internet location is not always allowed, that proxy settings HTTP_PROXY and HTTPS_PROXY need to be honored, and that custom CA trust needs to be taken into account. You can get a bundle of all public and additonal CAs from a configmap via auto-injection into the configmap data section [https://docs.openshift.com/container-platform/4.10/networking/configuring-a-custom-pki.html#certificate-injection-usin[…]rs_configuring-a-custom-pki](https://docs.openshift.com/container-platform/4.10/networking/configuring-a-custom-pki.html#certificate-injection-using-operators_configuring-a-custom-pki)

**Comment by manosnoam** [comment]
Verified in 2.4:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/9913945/a385d586-e068-4d78-ae17-d6e68aafa31f)
